File: algorithms/python/MergeTwoSortedLists.py
# @author jasper
# @create 2021-10-12 13:43

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = ListNode()
        pre = result
        while l1 and l2:
            if l1.val <= l2.val:
                result.next = l1
                l1 = l1.next
            else:
                result.next = l2
                l2 = l2.next
            result = result.next
        if l1:
            result.next = l1
        if l2:
            result.next = l2
        return pre.next

File: algorithms/java/src/MergeTwoSortedLists.java
/**
 * @author jasper
 * @create 2021-10-12 13:43
 */
public class MergeTwoSortedLists {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode head = new ListNode();
        ListNode p = head;
        while(l1 != null && l2!= null){
            if (l1.val < l2.val){
                head.next = l1;
                head = head.next;
                l1 = l1.next;
            } else {
                head.next = l2;
                head = head.next;
                l2 = l2.next;
            }
        }
        if (l1 != null){
            head.next = l1;
        }
        if (l2 != null){
            head.next = l2;
        }
        return p.next;
    }
}
